src/datacollector/DataCollectionManager.cs

```diff
@@ -0,0 +1,665 @@
+
+        /// <summary>
+        /// Attachment manager for performing file transfer from datacollector.
+        /// </summary>
+        private IDataCollectionAttachmentManager attachmentManager;
+
+        /// <summary>
+        /// Message sink for sending data collection messages to client..
+        /// </summary>
+        private IMessageSink messageSink;
+
+        /// <summary>
+        /// Events that can be subscribed by datacollectors.
+        /// </summary>
+        private TestPlatformDataCollectionEvents events;
+
+        /// <summary>
+        /// Specifies whether the object is disposed or not. 
+        /// </summary>
+        private bool disposed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataCollectionManager"/> class.
+        /// </summary>
+        internal DataCollectionManager() : this(new DataCollectionAttachmentManager(), new MessageSink())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataCollectionManager"/> class.
+        /// </summary>
+        /// <param name="datacollectionAttachmentManager">
+        /// The datacollection Attachment Manager.
+        /// </param>
+        /// <param name="messageSink">
+        /// The message Sink.
+        /// </param>
+        internal DataCollectionManager(IDataCollectionAttachmentManager datacollectionAttachmentManager, IMessageSink messageSink)
+        {
+            this.attachmentManager = datacollectionAttachmentManager;
+            this.messageSink = messageSink;
+            this.events = new TestPlatformDataCollectionEvents();
+
+            this.RunDataCollectors = new Dictionary<Type, DataCollectorWrapper>();
+        }
+
+        /// <summary>
+        /// Gets cache of data collectors associated with the run.
+        /// </summary>
+        internal Dictionary<Type, DataCollectorWrapper> RunDataCollectors { get; private set; }
+
+        /// <inheritdoc/>
+        public IDictionary<string, string> InitializeDataCollectors(string settingsXml)
+        {
+            ValidateArg.NotNull(settingsXml, "settingsXml");
+
+            var sessionId = new SessionId(Guid.NewGuid());
+            var dataCollectionContext = new DataCollectionContext(sessionId);
+            this.dataCollectionEnvironmentContext = DataCollectionEnvironmentContext.CreateForLocalEnvironment(dataCollectionContext);
+
+            this.attachmentManager.Initialize(sessionId, sourceDirectory, this.messageSink);
+
+            var executionEnvironmentVariables = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            var runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(settingsXml);
+            sourceDirectory = RunSettingsUtilities.GetTestResultsDirectory(runConfiguration);
+
+            var dataCollectionRunSettings = XmlRunSettingsUtilities.GetDataCollectionRunSettings(settingsXml);
```

Comment:
XmlRunSettingsUtilities, DataCollectionRunSettings does not have traces. Please add.